### PR TITLE
Fix includes for inclusion as an external project

### PR DESCRIPTION
**Pull request by Shillaker**
When using Faabric as an external project it can either be added with `add_subdirectory` in which case we need the public `target_include_directories` or as an installed library in which case the transitive header includes should still work.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,32 +13,37 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
+set(FAABRIC_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/include)
+
+# Output directories
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+# External libraries
+include(cmake/ExternalProjects.cmake)
+
```
Comment (Shillaker) [inline CMakeLists.txt:26]: Just moved this bit up to put all config at the top

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,32 +13,37 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
     add_library(${lib_name}_obj OBJECT ${lib_deps})
 
+    # Include directories
+    target_include_directories(${lib_name} 
+        PUBLIC ${FAABRIC_INCLUDE_DIR}
+        PUBLIC ${PISTACHE_INCLUDE_DIR}
+        PUBLIC ${SPDLOG_INCLUDE_DIR}
+        PUBLIC ${RAPIDJSON_INCLUDE_DIR}
+        )
+
```
Comment (Shillaker) [inline CMakeLists.txt:40]: This is the key change, rather than using a global `include_directories` for this repo

**Comment by csegarragonz** [comment]
Closes #43

**Review by csegarragonz — APPROVED** [review]

**Comment by csegarragonz** [comment]
@Shillaker I was trying to rebuild this myself, but the `examples` build seems to fail (albeit the tests "pass"). In fact, if you check the [latest action](https://github.com/faasm/faabric/pull/44/checks?check_run_id=1551196689), under the tab `Build Examples` you will see that the build step for the `server` fails.

To replicate just run the steps in the workflow file:
```bash
inv dev.cmake --shared
inv dev.cc faabric --shared
inv dev.install faabric --shared
inv examples
```

I will now have a look into why the tests actually pass.

**Review by csegarragonz — APPROVED** [review]

**Pull request merged**